examples/medplum-chart-demo/src/bots/core/general-encounter-note.ts

```diff
@@ -0,0 +1,227 @@
+  const user = medplum.getProfile() as Practitioner;
+
+  // Reason for visit is a required answer, so if it is missing throw an error
+  if (!answers['reason-for-visit']?.valueCoding) {
+    throw new Error('Must provide a reason for the visit');
+  }
+
+  // Parse the answers into more easily usable objects
+  const observationData: GeneralObservationData = {
+    bloodPressure: {
+      diastolic: answers['diastolic']?.valueInteger,
+      systolic: answers['systolic']?.valueInteger,
+    },
+    height: answers['height']?.valueQuantity,
+    weight: answers['weight']?.valueQuantity,
+    hotFlash: answers['hot-flashes']?.valueBoolean,
+    moodSwings: answers['mood-swings']?.valueBoolean,
+    vaginalDryness: answers['vaginal-dryness']?.valueBoolean,
+    sleepDisturbance: answers['sleep-disturbance']?.valueBoolean,
+    selfReportedHistory: answers['self-reported-history']?.valueString,
+    date: answers['date']?.valueDateTime,
+  };
+
+  if (observationData.height && observationData.weight) {
+    observationData.bmi = calculateBMI(observationData.height, observationData.weight);
+  }
+
+  const conditionData: ConditionData = {
+    reasonForVisit: answers['reason-for-visit']?.valueCoding,
+    problemList: answers['problem-list']?.valueBoolean || false,
+  };
+
+  const clinicalImpressionData: ClinicalImpressionData = {
+    assessment: answers['assessment']?.valueString,
+  };
+
+  // Take the objects and create bundle entries from the data for each resource
+  const partialObservations = createPartialGeneralObservations(observationData, generalCodes);
+  const observationEntries = createObservationEntries(observationData, encounter, user, partialObservations);
+  const conditionEntries = createConditionEntries(conditionData, encounter, user);
+  const clinicalImpressionEntries = createClinicalImpressionEntry(clinicalImpressionData, encounter, user);
+
+  // Create an array of all entries
+  const entry = [...observationEntries, ...conditionEntries];
+  if (clinicalImpressionEntries) {
+    entry.push(clinicalImpressionEntries);
+  }
+
+  // Create the bundle
+  const encounterNoteBatch: Bundle = {
+    resourceType: 'Bundle',
+    type: 'batch',
+    entry,
+  };
+
+  // Execute the batch to create all resources at once
+  const responseBundle = await medplum.executeBatch(encounterNoteBatch);
+  return responseBundle;
+}
+
+function createPartialGeneralObservations(
+  observationData: GeneralObservationData,
+  codes: Record<string, CodeableConcept>
+): Partial<Observation>[] {
+  const partials: Partial<Observation>[] = [];
+  for (const [key, value] of Object.entries(observationData)) {
+    if (!value || (key === 'bloodPressure' && !value.systolic && !value.diastolic) || key === 'date') {
+      continue;
+    }
+
+    const resource: Partial<Observation> = {
+      code:
+        key === 'selfReportedHistory' ? getSelfReportedCode(observationData.selfReportedHistory as string) : codes[key],
+    };
+
+    switch (key) {
```

Comment:
you can simplify these switch statements as follows: 

```ts
const observationTypes = {
'height': 'valueQuantity', 
'moodSwings': 'valueBoolean', 
/...
}

if(key !== bloodPressure) {
  resource[observationTypes[key]] = observationData[key]
}
  ```

Reply:
You can then just handle the case of the `bloodPressure` separately

Reply:
This also means you can combine all the `createPartial***Observations` functions into a single utility function. 


```ts
createObservations( 
  observationData, 
  codes, 
  observationTypes) {
   // handle blood pressure 
   // else
   // use `observationTypes` to fill `value**`
  // Fill in the default fields
  // return a full Observation
}

Reply:
This should really reduce the amount of specialized code per-note type